.circleci/config.yml

```diff
@@ -12,6 +12,34 @@ jobs:
       - run: sudo pip3 install invoke semver pyyaml
       - run: inv test
       - run: cp manifests/metallb.yaml manifests/metallb.yaml.prev
+  e2etest-bgp:
+    machine:
+      image: ubuntu-2004:202104-01
+    steps:
+      - checkout
+      - run: sudo apt-get update
+      - run: sudo apt-get install python3-pip
+      - run: sudo pip3 install invoke semver pyyaml
+      - run: GO111MODULE="on" go get sigs.k8s.io/kind@v0.10.0
+      - run: curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+      - run: sudo install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
+      - run: inv build
+      - run: inv dev-env -p bgp
+      - run: inv e2etest
```

Comment:
When I tested this locally, I got one failure because my cluster wasn't ready when i ran e2etest.

```
  May 21 17:05:20.701: Unexpected error:                                                                                      
      <*errors.errorString | 0xc000460490>: {                                                                                 
          s: "there are currently no ready, schedulable nodes in the cluster",                                                
      }                                                                                                                       
      there are currently no ready, schedulable nodes in the cluster                                                          
  occurred                                                                                                                    
```

I wonder if this could turn into random failures in CI?  Maybe there should be an explicit check of some kind that the cluster is ready to proceed with running the e2etest suite?

Reply:
Hi @russellb, we could add that. There are some checks in the test that wait for objects (e.g. service/pod) to be created so I am not sure why this triggered. I have been trying to reproduce with no success. I tried with `inv dev-env -p layer2 && inv e2etest` and `inv dev-env -p layer2 e2etest`, I also tried with BGP protocol. How did you run this?

Reply:
It was `inv dev-env -p bgp` and `inv e2etest` manually.  It passed the 2nd time I ran `e2etest`.  I haven't been able to reproduce it again though.  :(

Reply:
Ok, I reproduced it by running `docker image prune -a` before `inv dev-env -p bgp && inv e2etest`.  I think pulling images is making it take a bit longer to come up and that's why it failed my first try and not later tries to reproduce.

Reply:
I added a check in `tasks.py` when invoking `inv e2etest` that will wait for `condition=Ready` for all Pods in the `kube-system` and `metallb-system` namespaces (or timeout after 120s). It seems to work for me

Reply:
Thanks.  I see in the logs that when running `e2etest`, it does try to wait, but it times out after 2 minutes.

I was able to reproduce most of the time with:

```
inv dev-env-cleanup && docker image prune -a && inv dev-env -p bgp && inv e2etest
```

but haven't been able to reproduce again with the extra 2 minute wait you added in tasks.py.

Reply:
Ok, If it is timing out in the CI, I will increase it to 300 seconds